packages/features/settings/OutOfOfficeEntriesListForTeam.tsx

```diff
@@ -0,0 +1,280 @@
+import { keepPreviousData } from "@tanstack/react-query";
+import type { ColumnDef } from "@tanstack/react-table";
+import { Trans } from "next-i18next";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { useFormState } from "react-hook-form";
+
+import dayjs from "@calcom/dayjs";
+import { getUserAvatarUrl } from "@calcom/lib/getAvatarUrl";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc/react";
+import { Avatar, Button, DataTable, EmptyScreen, Icon, showToast, SkeletonText, Tooltip } from "@calcom/ui";
+import type { BookingRedirectForm } from "@calcom/web/pages/settings/my-account/out-of-office";
+
+interface OutOfOfficeEntry {
+  id: number;
+  uuid: string;
+  start: Date;
+  end: Date;
+  toUserId: number | null;
+  toUser: {
+    username: string;
+  } | null;
+  reason: {
+    id: number;
+    emoji: string;
+    reason: string;
+    userId: number;
+  } | null;
+  notes: string | null;
+  user: { id: number; avatarUrl: string; username: string; email: string } | null;
+}
+
+export const OutOfOfficeEntriesListForTeam = ({
+  editOutOfOfficeEntry,
+  teamOOOEntriesUpdated,
+}: {
+  editOutOfOfficeEntry: (entry: BookingRedirectForm) => void;
+  teamOOOEntriesUpdated: number;
+}) => {
+  const { t } = useLocale();
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [debouncedSearchTerm, setDebouncedSearchTerm] = useState("");
+  const [deletedEntry, setDeletedEntry] = useState(0);
+  const { data, isPending, fetchNextPage, isFetching, refetch } =
+    trpc.viewer.outOfOfficeEntriesList.useInfiniteQuery(
+      {
+        limit: 10,
+        fetchTeamMembersEntries: true,
+        searchTerm: debouncedSearchTerm,
+      },
+      {
+        getNextPageParam: (lastPage) => lastPage.nextCursor,
+        placeholderData: keepPreviousData,
+      }
+    );
+
+  useEffect(() => {
+    refetch();
+  }, [teamOOOEntriesUpdated, deletedEntry, refetch]);
+
+  const totalDBRowCount = data?.pages?.[0]?.meta?.totalRowCount ?? 0;
+  //Flatten the array of arrays from the useInfiniteQuery hook
+  const flatData = useMemo(
+    () => data?.pages?.flatMap((page) => page.rows) ?? [],
+    [data]
+  ) as OutOfOfficeEntry[];
+  const totalFetched = flatData.length;
+
```

Comment:
Hey, this is similar to the other component. I know there are a few different cases that don't match, but instead of doing the same things twice in two different components, can you please try to consolidate them into a single component? If things become too difficult and there are too many visible conditional changes, it's okay to define 2 components. However, in such cases, define things once and use them in another component. For example, EmptyScreen is defined twice, etc.

Reply:
handled in single component